torchgeo/datasets/nlcd.py

```diff
@@ -0,0 +1,310 @@
+        71: 12,
+        81: 13,
+        82: 14,
+        90: 15,
+        95: 16,
+    }
+
+    cmap = {
+        0: (0, 0, 0, 255),
+        1: (70, 107, 159, 255),
+        2: (209, 222, 248, 255),
+        3: (222, 197, 197, 255),
+        4: (217, 146, 130, 255),
+        5: (235, 0, 0, 255),
+        6: (171, 0, 0, 255),
+        7: (179, 172, 159, 255),
+        8: (104, 171, 95, 255),
+        9: (28, 95, 44, 255),
+        10: (181, 197, 143, 255),
+        11: (204, 184, 121, 255),
+        12: (223, 223, 194, 255),
+        13: (220, 217, 57, 255),
+        14: (171, 108, 40, 255),
+        15: (184, 217, 235, 255),
+        16: (108, 159, 184, 255),
+    }
+
+    def __init__(
+        self,
+        root: str = "data",
+        crs: Optional[CRS] = None,
+        res: Optional[float] = None,
+        years: List[int] = [2019],
+        transforms: Optional[Callable[[Dict[str, Any]], Dict[str, Any]]] = None,
+        cache: bool = True,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new Dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            crs: :term:`coordinate reference system (CRS)` to warp to
+                (defaults to the CRS of the first file found)
+            res: resolution of the dataset in units of CRS
+                (defaults to the resolution of the first file found)
+            years: list of years for which to use nlcd layer
+            transforms: a function/transform that takes an input sample
+                and returns a transformed version
+            cache: if True, cache file handle to speed up repeated sampling
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 after downloading files (may be slow)
+
+        Raises:
+            FileNotFoundError: if no files are found in ``root``
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+            AssertionError: if ``year`` is invalid
+        """
+        assert set(years).issubset(
+            self.md5s.keys()
+        ), f"NLCD data product only exists for the following years: {self.md5s.keys()}."
+        self.years = years
+        self.root = root
+        self.download = download
+        self.checksum = checksum
+
+        self._verify()
+
+        super().__init__(root, crs, res, transforms=transforms, cache=cache)
+
+    def __getitem__(self, query: BoundingBox) -> Dict[str, Any]:
+        """Retrieve image/mask and metadata indexed by query.
+
+        Args:
+            query: (minx, maxx, miny, maxy, mint, maxt) coordinates to index
+
+        Returns:
+            sample of image/mask and metadata at that index
+
+        Raises:
+            IndexError: if query is not found in the index
+        """
+        sample = super().__getitem__(query)
+
+        # replace mask values with ordinal values
```

Comment:
I wonder how much faster this is than a simple for-loop like we used in L7 Irish/L8 Biome. It's certainly less legible...

Reply:
I checked and it is actually not faster (current approach 0.011 sec mean across 100 samples, the simple for loop 0.004).